### PR TITLE
Release pipeline for GitHub actions

### PR DESCRIPTION
**Pull request by danielSanchezQ**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+          # Linux
+          - {os: ubuntu-latest, target: arm-unknown-linux-gnueabi, disable_tests: 1}
+          - {os: ubuntu-latest, target: armv7-unknown-linux-gnueabihf, disable_tests: 1}
+          - {os: ubuntu-latest, target: mips64el-unknown-linux-gnuabi64, disable_tests: 1}
+          - {os: ubuntu-latest, target: powerpc64le-unknown-linux-gnu, disable_tests: 1}
+          - {os: ubuntu-latest, target: x86_64-unknown-linux-musl, disable_tests: 0}
+          - {os: ubuntu-latest, target: aarch64-linux-android, disable_tests: 0}
+          # Macos
+          - {os: macos-latest, target: x86_64-unknown-linux-gnu, disable_tests: 0}
+          - {os: macos-latest, target: aarch64-unknown-linux-gnu, disable_tests: 1}
```
Comment (vincenthz) [inline .github/workflows/release.yml]: macos / linux on the same line looks fishy, is that really expected ?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      matrix:
+        config:
+          # Linux
+          - {os: ubuntu-latest, target: arm-unknown-linux-gnueabi, disable_tests: 1}
+          - {os: ubuntu-latest, target: armv7-unknown-linux-gnueabihf, disable_tests: 1}
+          - {os: ubuntu-latest, target: mips64el-unknown-linux-gnuabi64, disable_tests: 1}
+          - {os: ubuntu-latest, target: powerpc64le-unknown-linux-gnu, disable_tests: 1}
+          - {os: ubuntu-latest, target: x86_64-unknown-linux-musl, disable_tests: 0}
```
Comment (vincenthz) [inline .github/workflows/release.yml]: don't think we need a musl version

File: .github/workflows/release.yml (hunk 1 above)
Comment (danielSanchezQ) [reply .github/workflows/release.yml]: I took it from here: https://github.com/input-output-hk/jormungandr/blob/461fad847a30eb280d887b55ba790d45fa2d181f/.travis.yml#L18

File: .github/workflows/release.yml (hunk 2 above)
Comment (danielSanchezQ) [reply .github/workflows/release.yml]: It was also in the travis script: https://github.com/input-output-hk/jormungandr/blob/461fad847a30eb280d887b55ba790d45fa2d181f/.travis.yml#L27

File: .github/workflows/release.yml (hunk 1 above)
Comment (vincenthz) [reply .github/workflows/release.yml]: This line does create a linux env though, not a macos one as the `os` key is not set.

**Comment by danielSanchezQ** [comment]
It worked on my fork tests:
https://github.com/danielSanchezQ/jormungandr/runs/431774339?check_suite_focus=true
https://github.com/danielSanchezQ/jormungandr/releases/tag/v0.test_release3

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+      - name: Upload binaries to release
+        uses: svenstaro/upload-release-action@v1-release
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: release_test-${{ steps.get_version.outputs.VERSION }}-${{ matrix.config.target }}.*
+          asset_name: release_test-${{ steps.get_version.outputs.VERSION }}-${{ matrix.config.target }}
+          tag: ${{ github.ref }}
+          file_glob: true
+          overwrite: true
```
Comment (mzabaluev) [inline .github/workflows/release.yml]: Nitpick: missing newline

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    strategy:
+      matrix:
+        config:
+          # Linux
+          - {os: ubuntu-latest, target: arm-unknown-linux-gnueabi, disable_tests: 1}
+          - {os: ubuntu-latest, target: armv7-unknown-linux-gnueabihf, disable_tests: 1}
+          - {os: ubuntu-latest, target: mips64el-unknown-linux-gnuabi64, disable_tests: 1}
+          - {os: ubuntu-latest, target: powerpc64le-unknown-linux-gnu, disable_tests: 1}
+          - {os: ubuntu-latest, target: x86_64-unknown-linux-musl, disable_tests: 0}
+          - {os: ubuntu-latest, target: aarch64-linux-android, disable_tests: 0}
```
[the remark this answers is not in the document]
Comment (mzabaluev) [reply .github/workflows/release.yml]: Or, as no tests are actually run, just remove this flag for now.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+      - name: Checkout code
+        uses: actions/checkout@v1
+        with:
+          submodules: true
+
+      - name: Build jormungandr
+        uses: actions-rs/cargo@v1
+        env:
+          DISABLE_TESTS: ${{ matrix.config.disable_tests }}
```
[the remark this answers is not in the document]
Comment (danielSanchezQ) [reply .github/workflows/release.yml]: Will be removed with the others, but It was being set in travis:
https://github.com/input-output-hk/jormungandr/blob/461fad847a30eb280d887b55ba790d45fa2d181f/.travis.yml#L16

File: .github/workflows/release.yml (hunk 1 above)
Comment (danielSanchezQ) [reply .github/workflows/release.yml]: Ok, I misunderstood the Travis config then. Please confirm that the targets are ok. I changed the ones related to the comments below already.

**Comment by danielSanchezQ** [comment]
Assets are now properly named, and `.tar.gz` files are flat:
https://github.com/danielSanchezQ/jormungandr/actions/runs/35897872
https://github.com/danielSanchezQ/jormungandr/releases/tag/v0.test_release6

File: .github/workflows/release.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (vincenthz) [reply .github/workflows/release.yml]: I think @mzabaluev is right. the DISABLE_TESTS is used by the shell script only https://github.com/input-output-hk/jormungandr/blob/master/ci/script.sh#L10-L14 so it's quite likely we don't need this in the env

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        id: get_version
+        run: echo ::set-output name=VERSION::``${GITHUB_REF#refs/tags/}``
+
+      - name: Pack binaries if unix
+        if: matrix.config.os != 'windows-latest'
+        run: tar -C ./target/release -czvf jormungandr-${{ steps.get_version.outputs.VERSION }}-${{ matrix.config.target }}.tar.gz jormungandr jcli
+
+      - name: Pack binaries if windows
+        if: matrix.config.os == 'windows-latest'
+        run: compress-archive ./target/release/jormungandr.exe, ./target/release/jcli.exe jormungandr-${{ steps.get_version.outputs.VERSION }}-${{ matrix.config.target }}.zip
```
Comment (danielSanchezQ) [inline .github/workflows/release.yml]: For some reason, it is not taking the version for windows. I'm taking a look at this.

File: .github/workflows/release.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (danielSanchezQ) [reply .github/workflows/release.yml]: But should we still use it? I completly removed it now.

File: .github/workflows/release.yml (hunk 6 above)
Comment (danielSanchezQ) [reply .github/workflows/release.yml]: Ok, this is solved too.

Comment (danielSanchezQ) [reply .github/workflows/release.yml]: https://github.com/danielSanchezQ/jormungandr/runs/432047559?check_suite_focus=true
https://github.com/danielSanchezQ/jormungandr/releases/tag/v0.test_release7

File: .github/workflows/release.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (mzabaluev) [reply .github/workflows/release.yml]: This should be cleaned up, otherwise the config looks good to me.

**Pull request merged**